e2e/solve/test.go

```diff
@@ -383,12 +399,36 @@ func openOrder(ctx context.Context, backends ethbackend.Backends, order TestOrde
 	}, solvernet.WithFillDeadline(order.FillDeadline))
 }
 
-func testCheckAPI(ctx context.Context, orders []TestOrder) error {
+func testCheckAPI(ctx context.Context, backends ethbackend.Backends, orders []TestOrder) error {
 	const url = "http://localhost:26661/api/v1/check"
 
 	var eg errgroup.Group
-	for i, order := range orders {
+	var balanceChangingInProgress bool // Tracks if a solver account drain/refund cycle is running.
+	var refundMutex sync.Mutex
+	balanceRestored := sync.NewCond(&refundMutex)
```

Comment:
try to avoid complex async logic, this is very hard to follow and will probably flap. Rather make it synchronous and remove this.